vue/dynamicforms/src/components/api_view/detail-view-api.ts

```diff
@@ -4,57 +4,96 @@ import { computed, isRef, ref, Ref } from 'vue';
 import { APIConsumer } from '../api_consumer/namespace';
 import apiClient from '../util/api-client';
 
-import { IDetailViewApi, PrimaryKeyType, PrimaryKeyBaseType } from './namespace';
+import { IDetailViewApi, PrimaryKeyType, PrimaryKeyBaseType, QueryType } from './namespace';
 
 function urlParamToRef(url: string | Ref<string>) {
   const urlRef = isRef(url) ? url : ref(url);
   return computed(() => urlRef.value.replace(/\/+$/, ''));
 }
 
+function objectToURLEParam(obj: Partial<any> | Ref<Partial<any>>) {
+  const deRef: Partial<any> = isRef(obj) ? <Partial<any>> obj.value : obj;
+  return computed(() => new URLSearchParams(
+    Object.fromEntries(Object.entries(deRef).map(([k, v]) => [k, v.toString()])),
+  ));
+}
+
+function makeQuery(query?: QueryType): Ref<URLSearchParams> | undefined {
+  if (!query) return undefined;
+  if (isRef(query) && (query.value instanceof URLSearchParams)) return <Ref<URLSearchParams>> query;
+  if (isRef(query)) return objectToURLEParam(query);
+  if (query instanceof URLSearchParams) return ref(query);
+  return objectToURLEParam(query);
+}
+
 export default class DetailViewApi<T = any> implements IDetailViewApi<T> {
-  protected baseUrl: Ref<string>;
+  protected readonly baseUrl: Ref<string>;
+
+  protected readonly trailingSlash: boolean;
 
-  protected trailingSlash: boolean;
+  protected readonly pk?: Ref<PrimaryKeyBaseType>;
 
-  protected pk?: Ref<PrimaryKeyBaseType>;
+  protected readonly query?: Ref<URLSearchParams>;
 
-  private detail_url: Ref<string>;
+  public readonly detail_url: Ref<string>;
 
-  private definition_url: Ref<string>;
+  public readonly definition_url: Ref<string>;
 
-  private data_url: Ref<string>;
+  public readonly data_url: Ref<string>;
 
-  constructor(url: string | Ref<string>, trailingSlash: boolean = false, pk?: PrimaryKeyType) {
+  constructor(url: string | Ref<string>, trailingSlash: boolean = false, pk?: PrimaryKeyType, query?: QueryType) {
     this.baseUrl = urlParamToRef(url);
     this.trailingSlash = trailingSlash;
-    if (pk === undefined) this.pk = pk;
-    else this.pk = isRef(pk) ? pk : ref(pk);
+    if (pk === undefined) {
+      this.pk = pk;
+    } else {
+      this.pk = isRef(pk) ? pk : ref(pk);
```

Comment:
ref(pk). If is unecessery.

Reply:
not true. if I remove it, undefined will not be supported

Reply:
if you rewrite line 50 into: this.pk = ref(pk), it won't. It will do the exact same thing under the hood.